compiler-rt/lib/builtins/trampoline_setup.c

```diff
@@ -41,3 +41,30 @@ COMPILER_RT_ABI void __trampoline_setup(uint32_t *trampOnStack,
   __clear_cache(trampOnStack, &trampOnStack[10]);
 }
 #endif // __powerpc__ && !defined(__powerpc64__)
+
+// The AArch64 compiler generates calls to __trampoline_setup() when creating
+// trampoline functions on the stack for use with nested functions.
+// This function creates a custom 20-byte trampoline function on the stack
+// which loads x18 with a pointer to the outer function's locals
+// and then jumps to the target nested function.
+
+#if __aarch64__
```

Comment:
I would suggest adding `defined(__linux__)` guard as well.

Reply:
This can be `defined(__aarch64__) && defined(__ELF__)` since other ELF OSes likely need this as well.

Reply:
There is already an OS guard in the lowering code. Does it need to be done here as well? I don't think anything else call this code.

Reply:
I prefer the library implementation being guarded as well, so that its limitations are clear in the library source code.